workflow/lib/preprocess/preprocess.py

```diff
@@ -2,94 +2,111 @@
         combined_metadata["field_of_view"] = pd.to_numeric(
             combined_metadata["field_of_view"], errors="coerce"
         )
-        # Sort by 'field_of_view' in ascending order
-        combined_metadata = combined_metadata.sort_values("field_of_view").reset_index(
-            drop=True
-        )
-        return combined_metadata
-    else:
-        print("No valid ND2 files found in the provided list.")
-        return pd.DataFrame()  # Return an empty DataFrame if no files were processed
+        return combined_metadata.sort_values("field_of_view").reset_index(drop=True)
+    
+    print("No valid ND2 files found in the provided list.")
+    return pd.DataFrame()
 
 
-def nd2_to_tiff(file: str, channel_order_flip: bool = False) -> np.ndarray:
-    """Converts a single ND2 file with one field of view and multiple channels to a multidimensional numpy array.
+def nd2_to_tiff(file: str, channel_order_flip: bool = False, verbose: bool = False) -> np.ndarray:
+    """Converts a single ND2 file with one field of view and multiple channels to a multidimensional numpy array, 
+    ensuring the structure is CYX.
 
     Args:
         file (str): Path to the ND2 file.
-        channel_order_flip (bool, optional): If True, reverses the order of channels. Defaults to False.
+        channel_order_flip (bool, optional): If True, flips the channel order. Defaults to False.
+        verbose (bool, optional): If True, prints dimension information. Defaults to False.
 
     Returns:
         np.ndarray: Image data as a multidimensional numpy array.
     """
-    with ND2Reader(file) as images:
-        # Determine the axes order (always include 'c' for channels)
-        axes = "cyx"
-        if "z" in images.axes:
-            axes = "zcyx"
-
-        images.bundle_axes = axes
-
-        # Get the single image (all channels)
-        image = images[0]
-
-        # Handle z-stacks: max project if 'z' is in axes
-        if "z" in axes:
-            image = np.max(image, axis=0)  # Max projection along z-axis
-
-        # Flip channel order if specified
-        if channel_order_flip:
-            image = np.flip(image, axis=0)  # Flip along first axis (channels)
-
-        # Ensure the image is uint16
-        image_array = np.array(image, dtype=np.uint16)
-
-    return image_array
+    # Load with xarray to keep dimension labels
+    image = nd2.imread(file, xarray=True)
+    
+    if verbose:
+        print(f"Original dimensions: {image.dims}")
+       
+    # Handle Z-stack if present
+    if 'Z' in image.dims:
+        image = image.max(dim='Z')
+    
+    # Convert to numpy array in CYX order
+    image_array = image.transpose('C', 'Y', 'X').values
+
+    # Flip channel order on C axis if requested
+    if channel_order_flip:
+        image_array = np.flip(image_array, axis=0)  # axis 0 is the C axis
+
+    if verbose:
+        print(f"Final dimensions: {image_array.shape}")
+    
+    return image_array.astype(np.uint16)
```

Comment:
Blank line at the end of the file here. Using ruff format in Brieflow/ will help avoid these little things as well. You can also make python files [autoformat](https://stackoverflow.com/questions/50606758/vscode-how-do-you-autoformat-on-save) on save in vscode.

Reply:
Also changed. Running ruff is easy. Should we include it as a requirement to brieflow or is that not standard? I had to install in another conda env.